rust/operator-binary/src/csi_server/node.rs

```diff
@@ -242,15 +261,18 @@ impl SecretProvisionerNode {
         // however, we mostly just care about preventing accidental hashes here, for which plain byte truncation should be "good enough".
         let volume_tag = &volume_tag[..16];
 
-        let mut annotations = BTreeMap::default();
+        let mut annotations = Annotations::new();
 
         if let Some(expires_after) = data.expires_after {
             annotations.insert(
-                format!(
-                    "restarter.stackable.tech/expires-at.{:x}",
-                    FmtByteSlice(volume_tag)
-                ),
-                expires_after.to_rfc3339(),
+                Annotation::try_from((
+                    format!(
+                        "restarter.stackable.tech/expires-at.{:x}",
+                        FmtByteSlice(volume_tag)
+                    ),
+                    expires_after.to_rfc3339(),
+                ))
+                .context(publish_error::BuildAnnotationSnafu)?,
```

Comment:
As discussed, @Techassi: I have qualified the enum module here, but let me know if you want it imported.

Reply:
How are the other variants used in this file? I would suggest we use the same format here as other instances use.

Reply:
We should also think about adding a helper function to enable easier inserting of labels and annotations from raw K/v values.

Reply:
> How are the other variants used in this file? I would suggest we use the same format here as other instances use.

Yeah good point. They are the same.

https://github.com/stackabletech/secret-operator/blob/eb0d848e280542861d851a65925b9864f88b4c05/rust/operator-binary/src/csi_server/node.rs#L150-L153

Reply:
Okay, then we can leave it as is.